### ISSUE TITLE
openi_tracker failed to build for Ubuntu 13.04 (raring) 

### ISSUE THREAD

**Issue by Robbie025** [ros-drivers/openni_tracker#6]
It seems that the openi_tracker failed to build for Ubuntu 13.04 for hydro.

sudo apt-get install ros-hydro-openni-tracker

does not work and gives the following error 

Reading package lists... Done
Building dependency tree  
Reading state information... Done
E: Unable to locate package ros-hydro-openni-tracker

Thanks

**Comment by jonbinney** [comment]
Looks like it is failing because it can't find the dependency libopenni-nite-dev: 
http://jenkins.ros.org/view/HbinR64/job/ros-hydro-openni-tracker_binarydeb_raring_amd64/292/console

I think that libopenni-nite-dev is a package that gets manually copied to the ROS apt repo. @tfoote is that missing for raring?

**Comment by tfoote** [comment]
Yes, it's an externally provided binary that I don't know of a package
released for raring.

**Comment by bit-pirate** [comment]
AFAIK, this library was released by someone within Willow back. @tfoote do you know the maintainer and/or have access to those files/that library?

**Comment by tfoote** [comment]
The good news is I know who. The bad news is that I did much of that packing. The problem is that nite is not open source it's a binary blob. The interface is defined. I don't think the blob ports forward.  The debs are basically hand rolled. you could try just rewriting the metadata, but I'm pretty sure there's a reason it hasn't happened already.

**Comment by adwilson10** [comment]
@bit-pirate @tfoote  Myself and @jarvisschultz have been trying to figure out a way to get openni_tracker released into indigo and I think there are two options:
1.  We created a new deb package of libopenni-nite-dev for trusty based on the Nite 1.5.2.21 release.  As far as I can tell, the license restricting the redistribution of Nite only appears in 1.5.2.23 onward through 2.0.  So, we could get you versions of the debs for ubuntu if they could be hosted on packages.ros.org like libopenni-nite-dev_1.3.1.5 currently is.  
2.  Alternatively, it appears that openni_tracker can be built without any Nite headers or libraries.  So, if libopenni-nite-dev is removed as a dependency, and the lines to find Nite in CMakeLists.txt are removed, then openni_tracker can be built and distributed without Nite (it won't run successfully though unless Nite is manually installed after the fact).

If the second choice is better, I can create a pull request with the changes to openni_tracker to allow it to be built - either way, it may be worth removing these lines anyway from CMakeLists since it seems to have problems on other platforms, ie #7.

**Comment by tfoote** [comment]
If it can be achieved, I think the 2nd option is good. Especially if it can give a useful runtime error saying that nite cannot be found and link to where to get it.

**Comment by bit-pirate** [comment]
If we choose the 2nd option, then there should be an info (warning) during compile highlighting the need for a manual install of Nite. Also, (if possible) there should be a warning, if someone tries to run the tracker w/o nite being installed.
